lib/index.js

```diff
@@ -171,7 +171,9 @@ internals.implementation = function(server, options) {
               let err_message =
                 verify_err.message === 'jwt expired'
                   ? 'Expired token'
-                  : 'Invalid token';
+                  : verify_err.message === 'invalid signature'
+                    ? null
```

Comment:
@ablackbu why `null` ?

Reply:
So after further investigation, I think it is a bug in hapi. It will return a 200 for optional even if it is not validated. 

If I put a breakpoint on the failing test at this part of hapi. We see a very interesting result.
https://github.com/hapijs/hapi/blob/v17.5.2/lib/server.js#L309

res.request.auth has: 
`{"data":null,"isBoom":true,"isServer":false,"output":{"statusCode":401,"payload":{"statusCode":401,"error":"Unauthorized","message":"Missing authentication"},"headers":{"WWW-Authenticate":"Token"}}}
}`

res has:
`
result: "worked",
statusCode: 200,
statusMessage: "OK"
`

 Thoughts?  Seems like a hapi bug around the optional parameter. It is saying its ok even if it is not.